src/components/LandEnsPage/LandEnsForm/LandEnsForm.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react'
+import { Form, Row, Button, Loader } from 'decentraland-ui'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { Link } from 'react-router-dom'
+import { locations } from 'routing/locations'
+import { Props, State } from './LandEnsForm.types'
+import './LandEnsForm.css'
+import { marketplace } from 'lib/api/marketplace'
+import { store } from 'modules/common/store'
+import { getAddress } from 'decentraland-dapps/dist/modules/wallet/selectors'
+import { env } from 'decentraland-commons'
+import { SelectNames } from 'components/SelectNames'
+
+export const ENS_CONTRACT_ADDRESS = env.get('REACT_APP_ENS_CONTRACT_ADDRESS', '')
+export const RESOLVER_CONTRACT_ADDRESS = env.get('REACT_APP_RESOLVER_CONTRACT_ADDRESS', '')
+
+export default class LandEnsForm extends React.PureComponent<Props, State> {
+  state: State = {
+    selectedSubdomain: '',
+    subdomainList: [],
+    done: false
+  }
+
+  componentDidMount() {
+    (async () => {
+     const state = store.getState()
+     const address = getAddress(state)
+     
+     if (address) {
+       const subdomainList = await marketplace.fetchDomainList(address)
+       this.setState({subdomainList})
+     }
+    })()
+  }
+    
+  handleChange = (selectedSubdomain: string) => {
+    const { onGetENS, land } = this.props
+    onGetENS(selectedSubdomain, land)
+    this.setState({ selectedSubdomain })
+  }
+
+  handleSubmit = () => {
+    const { onSetENS, land } = this.props
+    const { selectedSubdomain } = this.state
+    onSetENS(selectedSubdomain, land)
+    this.setState({done: true})
+  }
+  
+  render() {
+    const { land, ens, isLoading, error } = this.props
+    const { subdomainList, selectedSubdomain, done } = this.state
+    const selectOptions = subdomainList.map(x => ({value: x.toLowerCase(), text: x.toLowerCase()}))
```

Comment:
Nitpick but, maybe we can rename `x` to `subdomain` to be a bit more explicit
What do you think?

Reply:
Totally agree. Fixed